testsuite/MDAnalysisTests/lib/test_distances.py

```diff
@@ -264,24 +273,61 @@ def _dist(x, ref):
+        _, all, ref, _ = request.getfixturevalue(pos)
 
+        d = distances.distance_array(ref, all, backend=backend)
         assert_almost_equal(d, np.array([[
-            self._dist(points[0], ref[0]),
-            self._dist(points[1], ref[0]),
-            self._dist(points[2], ref[0]),
-            self._dist(points[3], ref[0])]
+            self._dist(points[0], reference[0]),
+            self._dist(points[1], reference[0]),
+            self._dist(points[2], reference[0]),
+            self._dist(points[3], reference[0])]
         ]))
 
-    def test_PBC(self, backend, ref_system):
-        box, points, ref, conf = ref_system
-
-        d = distances.distance_array(ref, points, box=box, backend=backend)
+    # cycle through combinations of numpy array and AtomGroup
+    @pytest.mark.parametrize('pos0', ['ref_system', 'ref_system_universe'])
+    @pytest.mark.parametrize('pos1', ['ref_system', 'ref_system_universe'])
+    def test_noPBC_mixed_combinations(self, backend, ref_system, pos0, pos1,
+                                      request):
+        _, points, reference, _ = ref_system  # reference values
+        _, _, ref_val, _ = request.getfixturevalue(pos0)
+        _, points_val, _, _ = request.getfixturevalue(pos1)
+        d = distances.distance_array(ref_val, points_val,
+                                     backend=backend)
+        assert_almost_equal(d, np.array([[
+            self._dist(points[0], reference[0]),
+            self._dist(points[1], reference[0]),
+            self._dist(points[2], reference[0]),
+            self._dist(points[3], reference[0])]
+        ]))
 
-        assert_almost_equal(d, np.array([[0., 0., 0., self._dist(points[3], ref=[1, 1, 2])]]))
+    # test both AtomGroup and numpy array
+    @pytest.mark.parametrize('pos', ['ref_system', 'ref_system_universe'])
+    def test_PBC(self, backend, ref_system, pos, request):
+        box, points, _, _ = ref_system
+        _, all, ref, _ = request.getfixturevalue(pos)
+
+        d = distances.distance_array(ref, all, box=box, backend=backend)
+
+        assert_almost_equal(d, np.array([[0., 0., 0., self._dist(points[3],
+                            ref=[1, 1, 2])]]))
+
+    # cycle through combinations of numpy array and AtomGroup
+    @pytest.mark.parametrize('pos0', ['ref_system', 'ref_system_universe'])
+    @pytest.mark.parametrize('pos1', ['ref_system', 'ref_system_universe'])
+    def test_PBC_mixed_combinations(self, backend, ref_system, pos0, pos1,
+                                    request):
+        box, points, _, _ = ref_system
+        _, _, ref_val, _ = request.getfixturevalue(pos0)
+        _, points_val, _, _ = request.getfixturevalue(pos1)
+        d = distances.distance_array(ref_val, points_val,
+                                     box=box,
+                                     backend=backend)
+        assert_almost_equal(d, np.array([[0., 0., 0., self._dist(points[3],
+                            ref=[1, 1, 2])]]))
```

Comment:
This would probably fit visual indentation and improve readability.

```suggestion
        assert_almost_equal(
            d, np.array([[0., 0., 0., self._dist(points[3], ref=[1, 1, 2])]]))
```